osbuild/util/containers.py

```diff
@@ -127,4 +148,32 @@ def container_source(image):
             os.symlink(image_filepath, tmp_source)
 
         image_source = f"{container_format}:{tmp_source}"
-        yield image_name, image_source
+        yield image_source
+
+
+@contextmanager
+def container_source(image):
+    image_filepath = image["filepath"]
+    container_format = image["data"]["format"]
+    image_name = image["data"]["name"]
+
+    if container_format not in ("dir", "oci-archive", "containers-storage"):
```

Comment:
(super nitpick) we could make this check part of the if/elif below. The (tiny) advantage would be that if we ever add a new format we only need to change it in one place (inside the if) but it's a super nitpick, my idea would look like this:
```python
diff --git a/osbuild/util/containers.py b/osbuild/util/containers.py
index da7ce081..480cd14f 100644
--- a/osbuild/util/containers.py
+++ b/osbuild/util/containers.py
@@ -157,15 +157,14 @@ def container_source(image):
     container_format = image["data"]["format"]
     image_name = image["data"]["name"]
 
-    if container_format not in ("dir", "oci-archive", "containers-storage"):
-        raise RuntimeError(f"Unknown container format {container_format}")
-
     if container_format == "containers-storage":
         with containers_storage_source(image, image_name, image_filepath, container_format) as image_source:
             yield image_name, image_source
-    else:
+    elif container_format in ("dir", "oci-archive"):
         with dir_oci_archive_source(image, image_name, image_filepath, container_format) as image_source:
             yield image_name, image_source
+    else:
+      raise RuntimeError(f"Unknown container format {container_format}")
 
 
 def get_host_storage(storage_conf):
```
but totally optional and nitpicky

Reply:
Oh yeah I like this